pacakges/server/src/core.rs

```diff
@@ -124,37 +126,108 @@ pub struct Requirement{
     pub message: Option<String>,
 }
 
+impl Requirement {
+    fn create_requirement(bank_name: String, credit_requirment: f32) -> Requirement {
+        Requirement {
+            course_bank_name: bank_name,
+            credit_requirment,
+            credit_complete: 0.0,
+            message: None,
+        }
+    }
+    fn with_credits(mut self, credit_complete: f32) -> Requirement {
+        self.credit_complete = credit_complete;
+        self
+    }
+    fn add_message(mut self, msg: String) -> Requirement {
```

Comment:
change both methods to with_ prefix or add_ prefix for consistency.
personally, I prefer "with_" prefix for both